code/datums/diseases/advance/symptoms/heal.dm

```diff
@@ -503,24 +521,59 @@
 	if(A.totalTransmittable() >= 6)
 		temp_rate = 4
 
+// We do this to prevent liver damage from injecting plasma when plasma fixation virus reaches stage 4 and beyond
+/datum/symptom/heal/plasma/on_stage_change(datum/disease/advance/A)
+	. = ..()
+	if(!.)
+		return FALSE
+		
+	if(A.stage >= 4)
+		ADD_TRAIT(A.affected_mob, TRAIT_PLASMA_LOVER_METABOLISM, DISEASE_TRAIT)
+	else 
+		REMOVE_TRAIT(A.affected_mob, TRAIT_PLASMA_LOVER_METABOLISM, DISEASE_TRAIT)
+	return TRUE
+
+/datum/symptom/heal/plasma/End(datum/disease/advance/A)
+	. = ..()
+	if(!.)
+		return
+		
+	REMOVE_TRAIT(A.affected_mob, TRAIT_PLASMA_LOVER_METABOLISM, DISEASE_TRAIT)
```

Comment:
Again, no single letter vars.

Reply:
This is how all of the virology code does it. Do you want me to refactor all of it?

Reply:
We're trying to slowly work away single letter vars. It might seem inconsistent for now, but it's the lesser evil
https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#dont-use-abbreviations

Reply:
Gotcha--I will make the change. I'll let someone else worry about refactoring the whole thing.